strimzi-operator/src/main/java/io/spoud/kcc/operator/users/KafkaUserService.java

```diff
@@ -41,11 +44,17 @@ public Collection<KafkaUser> getWritersOfTopic(String topicName) {
     }
 
     private boolean canPerformOperationOnTopic(KafkaUser user, AclOperation op, String topicName) {
-        if (user.getSpec().getAuthorization() instanceof KafkaUserAuthorizationSimple auth) {
-            var rules = getRulesForTopicName(auth.getAcls(), topicName);
-            // deny rule takes precedence over allow rule, so we need to check that there is no deny rule
-            return hasRuleForOperation(rules, op, AclRuleType.ALLOW) &&
-                    !hasRuleForOperation(rules, op, AclRuleType.DENY);
+        try {
+            if (user.getSpec().getAuthorization() instanceof KafkaUserAuthorizationSimple auth) {
+                var rules = getRulesForTopicName(auth.getAcls(), topicName);
+                // deny rule takes precedence over allow rule, so we need to check that there is no deny rule
+                return hasRuleForOperation(rules, op, AclRuleType.ALLOW) &&
+                        !hasRuleForOperation(rules, op, AclRuleType.DENY);
+            }
+        } catch (Exception e) {
+            var err = String.format("Error while checking ACLs for user %s. Context generation will proceed, but the user will not be listed as a reader/writer.",
+                    user.getMetadata().getName());
+            Log.warn(err, e);
```

Comment:
```suggestion
            Log.warnf(err, "Error while checking ACLs for user %s. Context generation will proceed, but the user will not be listed as a reader/writer.", user.getMetadata().getName());
```
Never do it manually, the logger is smart enough to decide when to do the format or not. Here, it's a WARN so I guess it would anyway be evaluated 100% of the time, but for DEBUG for example, you never know.